libraries/GCS_MAVLink/GCS_Signing.cpp

```diff
@@ -143,7 +164,7 @@ void GCS_MAVLINK::load_signing_key(void)
     // prevents a window for replay attacks
     signing.timestamp = key.timestamp + 60UL * 100UL * 1000UL;
     signing.flags = MAVLINK_SIGNING_FLAG_SIGN_OUTGOING;
-    signing.accept_unsigned_callback = accept_unsigned_callback;
+    signing.accept_unsigned_callback = generate_accept_unsigned_callback(option_enabled(Option::MAVLINK2_SIGNING_DISABLED));
```

Comment:
I don't think you need the `generate_` method here...

Just have an if/else right here?